lib/run_loop/life_cycle/core_simulator.rb

```diff
@@ -453,11 +453,21 @@ def wait_for_device_state(target_state)
       end
 
       # @!visibility private
+      def device_caches_dir
+        @device_caches_dir ||= File.join(device_data_dir, 'Library', 'Caches')
+      end
+
+      # @!visibility private
       def reset_app_sandbox_internal_shared
         [app_documents_dir, app_tmp_dir].each do |dir|
           FileUtils.rm_rf dir
           FileUtils.mkdir dir
         end
+        csstore_regex = File.join(device_caches_dir, "com.apple.LaunchServices-*.csstore")
+        matches = Dir[csstore_regex]
```

Comment:
I think we should delete all matches and avoid the if:
matches.each {|match| FileUtils.rm_f match}